src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/Endpoints/RazorCustomMessageTarget_Completion.cs

```diff
@@ -109,7 +110,11 @@ internal partial class RazorCustomMessageTarget
             await _joinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
             var provisionalChange = new VisualStudioTextChange(provisionalTextEdit, virtualDocumentSnapshot.Snapshot);
-            UpdateVirtualDocument(provisionalChange, request.ProjectedKind, request.Identifier.Version, hostDocumentUri, virtualDocumentSnapshot.Uri);
+            // We update to a negative version number so that if a request comes in for v6, it won't see our modified document. We revert the version back
+            // later, don't worry.
+            UpdateVirtualDocument(provisionalChange, request.ProjectedKind, -1 * request.Identifier.Version, hostDocumentUri, virtualDocumentSnapshot.Uri);
```

Comment:
maybe a pathological case,  but for small versions is there any concern? E.g we're setting this to -1 for v1. Could updates cause some sort of collision there? Should this just be `int.MinValue + request.Identifier.Version`?

Reply:
So it needs to be less than any other version, so that when something comes in it doesn't think "oh, there is a newer version, I'll cancel." It could be MinValue, but I thought it actually looks nice in the logs that you can see which version caused the provisional update.

The only possible conflict would be two provisional completions at the same time but were protected by the UI thread from that.


Oh and not to toot my own horn but this situation is impossible to hit in cohosting 😝